python2/typing.py

```diff
@@ -281,6 +290,14 @@ def __getitem__(self, parameter):
         return self.__class__(self.name, parameter,
                               self.impl_type, self.type_checker)
 
+    def __eq__(self, other):
+        if not isinstance(other, _TypeAlias):
+            return NotImplemented
+        return self.name == other.name and self.type_var == other.type_var
```

Comment:
This is an internal class, we know that if ``name`` is the same, then ``impl_type`` and ``type_checker`` must also be the same.